<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to arbalistIO • arbalistIO</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to arbalistIO">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">arbalistIO</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/basic.html">Introduction to arbalistIO</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/arbalist-inc/arbalistIO/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to arbalistIO</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/arbalist-inc/arbalistIO/blob/main/vignettes/basic.Rmd" class="external-link"><code>vignettes/basic.Rmd</code></a></small>
      <div class="hidden name"><code>basic.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p><strong>arbalistIO</strong> facilitates the import of single-cell
ATAC-seq and Multiome (ATAC + Gene Expression) data into Bioconductor’s
<code>MultiAssayExperiment</code>. It provides efficient C++ based
parsers for 10x Genomics fragment files to generate HDF5-backed sparse
matrices for both genomic tiles and custom regions.</p>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/arbalist-inc/arbalistIO" class="external-link">arbalistIO</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://waldronlab.io/MultiAssayExperiment/" class="external-link">MultiAssayExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Huber-group-EMBL/rhdf5" class="external-link">rhdf5</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://Matrix.R-forge.R-project.org" class="external-link">Matrix</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="preparing-mock-data">1. Preparing Mock Data<a class="anchor" aria-label="anchor" href="#preparing-mock-data"></a>
</h2>
<p>For this vignette, we will mock data that simulates the output of 10x
Genomics Cell Ranger. This includes a fragment file (ATAC-seq) and a
filtered feature matrix (Gene Expression).</p>
<div class="section level3">
<h3 id="generate-a-fragment-file">Generate a Fragment File<a class="anchor" aria-label="anchor" href="#generate-a-fragment-file"></a>
</h3>
<p>We use <code>mockFragmentFile</code> to create a fragment file
containing reads for 50 cells.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">fragment_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">tmp_dir</span>, <span class="st">"fragments.tsv.gz"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seq_lengths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>chr1 <span class="op">=</span> <span class="fl">10000</span>, chr2 <span class="op">=</span> <span class="fl">5000</span><span class="op">)</span></span>
<span><span class="va">cell_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Cell"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/mockFragmentFile.html">mockFragmentFile</a></span><span class="op">(</span></span>
<span>  output.file <span class="op">=</span> <span class="va">fragment_file</span>,</span>
<span>  seq.lengths <span class="op">=</span> <span class="va">seq_lengths</span>,</span>
<span>  num.fragments <span class="op">=</span> <span class="fl">5000</span>,</span>
<span>  cell.names <span class="op">=</span> <span class="va">cell_names</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="generate-a-mock-feature-matrix-hdf5">Generate a Mock Feature Matrix (HDF5)<a class="anchor" aria-label="anchor" href="#generate-a-mock-feature-matrix-hdf5"></a>
</h3>
<p><code>createArbalistMAE</code> requires the Cell Ranger filtered
feature matrix (HDF5) to identify valid barcodes.</p>
<p>TODO: add this as a function to the package.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h5_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">tmp_dir</span>, <span class="st">"filtered_feature_bc_matrix.h5"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">h5_file</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="va">h5_file</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5_createFile.html" class="external-link">h5createFile</a></span><span class="op">(</span><span class="va">h5_file</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5_createGroup.html" class="external-link">h5createGroup</a></span><span class="op">(</span><span class="va">h5_file</span>, <span class="st">"matrix"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5_createGroup.html" class="external-link">h5createGroup</a></span><span class="op">(</span><span class="va">h5_file</span>, <span class="st">"matrix/features"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">barcodes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">cell_names</span>, <span class="st">"-1"</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5_write.html" class="external-link">h5write</a></span><span class="op">(</span><span class="va">barcodes</span>, <span class="va">h5_file</span>, <span class="st">"matrix/barcodes"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">n_genes</span> <span class="op">&lt;-</span> <span class="fl">10</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5_write.html" class="external-link">h5write</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"GENE"</span>, <span class="fl">1</span><span class="op">:</span><span class="va">n_genes</span><span class="op">)</span>, <span class="va">h5_file</span>, <span class="st">"matrix/features/id"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5_write.html" class="external-link">h5write</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Gene"</span>, <span class="fl">1</span><span class="op">:</span><span class="va">n_genes</span><span class="op">)</span>, <span class="va">h5_file</span>, <span class="st">"matrix/features/name"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5_write.html" class="external-link">h5write</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Gene Expression"</span>, <span class="va">n_genes</span><span class="op">)</span>, <span class="va">h5_file</span>, <span class="st">"matrix/features/feature_type"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5_write.html" class="external-link">h5write</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"chr1:1-100"</span>, <span class="va">n_genes</span><span class="op">)</span>, <span class="va">h5_file</span>, <span class="st">"matrix/features/interval"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5_write.html" class="external-link">h5write</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Genome1"</span>, <span class="va">n_genes</span><span class="op">)</span>, <span class="va">h5_file</span>, <span class="st">"matrix/features/genome"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/Matrix.html" class="external-link">Matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Poisson.html" class="external-link">rpois</a></span><span class="op">(</span><span class="va">n_genes</span> <span class="op">*</span> <span class="fl">50</span>, lambda <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="va">n_genes</span>, sparse <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5_write.html" class="external-link">h5write</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">counts</span><span class="op">@</span><span class="va">x</span><span class="op">)</span>, <span class="va">h5_file</span>, <span class="st">"matrix/data"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5_write.html" class="external-link">h5write</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">counts</span><span class="op">@</span><span class="va">i</span><span class="op">)</span>, <span class="va">h5_file</span>, <span class="st">"matrix/indices"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5_write.html" class="external-link">h5write</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">counts</span><span class="op">@</span><span class="va">p</span><span class="op">)</span>, <span class="va">h5_file</span>, <span class="st">"matrix/indptr"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5_write.html" class="external-link">h5write</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>, <span class="va">h5_file</span>, <span class="st">"matrix/shape"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="creating-the-multiassayexperiment">2. Creating the MultiAssayExperiment<a class="anchor" aria-label="anchor" href="#creating-the-multiassayexperiment"></a>
</h2>
<p>Now we can use <code>createArbalistMAE</code> to import everything
into a single object. We will also define a set of genomic regions
(e.g., promoters) to quantify accessibility alongside the standard tile
matrix.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">promoters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span></span>
<span>  seqnames <span class="op">=</span> <span class="st">"chr1"</span>,</span>
<span>  ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">1000</span>, by <span class="op">=</span> <span class="fl">200</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">mae</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createArbalistMAE.html">createArbalistMAE</a></span><span class="op">(</span></span>
<span>  sample.names <span class="op">=</span> <span class="st">"MockSample"</span>,</span>
<span>  fragment.files <span class="op">=</span> <span class="va">fragment_file</span>,</span>
<span>  filtered.feature.matrix.files <span class="op">=</span> <span class="va">h5_file</span>,</span>
<span>  output.dir <span class="op">=</span> <span class="va">tmp_dir</span>,</span>
<span>  gene.grs <span class="op">=</span> <span class="va">promoters</span>,</span>
<span>  tile.size <span class="op">=</span> <span class="fl">500</span>,</span>
<span>  seq.lengths <span class="op">=</span> <span class="va">seq_lengths</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">mae</span><span class="op">)</span></span>
<span><span class="co">#&gt; A MultiAssayExperiment object of 3 listed</span></span>
<span><span class="co">#&gt;  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">#&gt;  Containing an ExperimentList class object of length 3:</span></span>
<span><span class="co">#&gt;  [1] TileMatrix500: SingleCellExperiment with 30 rows and 50 columns</span></span>
<span><span class="co">#&gt;  [2] GeneAccessibilityMatrix: SingleCellExperiment with 5 rows and 50 columns</span></span>
<span><span class="co">#&gt;  [3] GeneExpressionMatrix: SingleCellExperiment with 10 rows and 50 columns</span></span>
<span><span class="co">#&gt; Functionality:</span></span>
<span><span class="co">#&gt;  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">#&gt;  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">#&gt;  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">#&gt;  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">#&gt;  exportClass() - save data to flat files</span></span></code></pre></div>
<p>The resulting <code>MultiAssayExperiment</code> contains three
experiments:</p>
<ol style="list-style-type: decimal">
<li>
<strong>TileMatrix500</strong>: 500bp tile counts (ATAC).</li>
<li>
<strong>GeneAccessibilityMatrix</strong>: Counts over the provided
<code>gene.grs</code> regions (ATAC).</li>
<li>
<strong>GeneExpressionMatrix</strong>: RNA counts imported from the
HDF5 file.</li>
</ol>
<div class="section level3">
<h3 id="exploring-the-object">Exploring the Object<a class="anchor" aria-label="anchor" href="#exploring-the-object"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://github.com/waldronlab/MultiAssayExperiment/reference/MultiAssayExperiment-methods.html" class="external-link">experiments</a></span><span class="op">(</span><span class="va">mae</span><span class="op">)</span><span class="op">[[</span><span class="st">"TileMatrix500"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">tiles</span><span class="op">)</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 30 50 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames: NULL</span></span>
<span><span class="co">#&gt; rowData names(0):</span></span>
<span><span class="co">#&gt; colnames(50): MockSample#Cell1-1 MockSample#Cell2-1 ...</span></span>
<span><span class="co">#&gt;   MockSample#Cell49-1 MockSample#Cell50-1</span></span>
<span><span class="co">#&gt; colData names(1): Sample</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: TileMatrix500</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span></span>
<span><span class="va">rna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://github.com/waldronlab/MultiAssayExperiment/reference/MultiAssayExperiment-methods.html" class="external-link">experiments</a></span><span class="op">(</span><span class="va">mae</span><span class="op">)</span><span class="op">[[</span><span class="st">"GeneExpressionMatrix"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">rna</span><span class="op">)</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 10 50 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames: NULL</span></span>
<span><span class="co">#&gt; rowData names(5): feature_type genome id interval name</span></span>
<span><span class="co">#&gt; colnames(50): MockSample#Cell1-1 MockSample#Cell2-1 ...</span></span>
<span><span class="co">#&gt;   MockSample#Cell49-1 MockSample#Cell50-1</span></span>
<span><span class="co">#&gt; colData names(1): Sample</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: GeneExpressionMatrix</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">mae</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 1 row and 1 column</span></span>
<span><span class="co">#&gt;                     fragment_file</span></span>
<span><span class="co">#&gt;                       &lt;character&gt;</span></span>
<span><span class="co">#&gt; MockSample /tmp/RtmpCKJcMq/frag..</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="low-level-matrix-generation">3. Low-Level Matrix Generation<a class="anchor" aria-label="anchor" href="#low-level-matrix-generation"></a>
</h2>
<p>If you do not need the full <code>MultiAssayExperiment</code>
structure, you can use the underlying C++ wrappers to generate specific
matrices.</p>
<div class="section level3">
<h3 id="generating-a-tile-matrix">Generating a Tile Matrix<a class="anchor" aria-label="anchor" href="#generating-a-tile-matrix"></a>
</h3>
<p>Use <code>saveTileMatrix</code> to bin fragments into genome-wide
tiles.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tile_h5</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">tmp_dir</span>, <span class="st">"my_tiles.h5"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tile_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/saveTileMatrix.html">saveTileMatrix</a></span><span class="op">(</span></span>
<span>  fragment.file <span class="op">=</span> <span class="va">fragment_file</span>,</span>
<span>  output.file <span class="op">=</span> <span class="va">tile_h5</span>,</span>
<span>  output.name <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  seq.lengths <span class="op">=</span> <span class="va">seq_lengths</span>,</span>
<span>  tile.size <span class="op">=</span> <span class="fl">1000</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Returns a list with the GRanges tiles and the HDF5Matrix</span></span>
<span><span class="va">tile_res</span><span class="op">$</span><span class="va">tiles</span></span>
<span><span class="co">#&gt; GRanges object with 15 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;        seqnames    ranges strand</span></span>
<span><span class="co">#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;    [1]     chr1    1-1000      *</span></span>
<span><span class="co">#&gt;    [2]     chr1 1001-2000      *</span></span>
<span><span class="co">#&gt;    [3]     chr1 2001-3000      *</span></span>
<span><span class="co">#&gt;    [4]     chr1 3001-4000      *</span></span>
<span><span class="co">#&gt;    [5]     chr1 4001-5000      *</span></span>
<span><span class="co">#&gt;    ...      ...       ...    ...</span></span>
<span><span class="co">#&gt;   [11]     chr2    1-1000      *</span></span>
<span><span class="co">#&gt;   [12]     chr2 1001-2000      *</span></span>
<span><span class="co">#&gt;   [13]     chr2 2001-3000      *</span></span>
<span><span class="co">#&gt;   [14]     chr2 3001-4000      *</span></span>
<span><span class="co">#&gt;   [15]     chr2 4001-5000      *</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome</span></span>
<span><span class="va">tile_res</span><span class="op">$</span><span class="va">counts</span></span>
<span><span class="co">#&gt; &lt;15 x 50&gt; sparse DelayedMatrix object of type "integer":</span></span>
<span><span class="co">#&gt;        Cell2  Cell5 Cell22 Cell15 ... Cell43 Cell35 Cell46 Cell28</span></span>
<span><span class="co">#&gt;  [1,]    238    217    236    243   .    262    247    247    237</span></span>
<span><span class="co">#&gt;  [2,]    353    341    374    387   .    372    394    395    415</span></span>
<span><span class="co">#&gt;  [3,]    377    412    386    350   .    364    367    399    405</span></span>
<span><span class="co">#&gt;  [4,]    397    412    399    358   .    404    395    342    408</span></span>
<span><span class="co">#&gt;  [5,]    398    379    380    393   .    388    399    353    382</span></span>
<span><span class="co">#&gt;   ...      .      .      .      .   .      .      .      .      .</span></span>
<span><span class="co">#&gt; [11,]    495    514    500    486   .    452    475    497    460</span></span>
<span><span class="co">#&gt; [12,]    752    775    745    741   .    722    768    739    709</span></span>
<span><span class="co">#&gt; [13,]    749    760    763    724   .    718    764    747    704</span></span>
<span><span class="co">#&gt; [14,]    746    779    754    745   .    769    763    745    776</span></span>
<span><span class="co">#&gt; [15,]    759    730    764    779   .    794    758    737    763</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="generating-a-region-matrix">Generating a Region Matrix<a class="anchor" aria-label="anchor" href="#generating-a-region-matrix"></a>
</h3>
<p>Use <code>saveRegionMatrix</code> to count fragments overlapping
specific genomic regions</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">region_h5</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">tmp_dir</span>, <span class="st">"my_regions.h5"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Define arbitrary regions</span></span>
<span><span class="va">peaks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">5000</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">200</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">region_mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/saveRegionMatrix.html">saveRegionMatrix</a></span><span class="op">(</span></span>
<span>  fragment.file <span class="op">=</span> <span class="va">fragment_file</span>,</span>
<span>  output.file <span class="op">=</span> <span class="va">region_h5</span>,</span>
<span>  output.name <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  regions <span class="op">=</span> <span class="va">peaks</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">region_mat</span><span class="op">)</span></span>
<span><span class="co">#&gt; &lt;2 x 50&gt; sparse DelayedMatrix object of type "raw":</span></span>
<span><span class="co">#&gt;       Cell2  Cell5 Cell22 ... Cell46 Cell28</span></span>
<span><span class="co">#&gt; [1,]     37     28     2b   .     31     2c</span></span>
<span><span class="co">#&gt; [2,]     5e     68     58   .     58     56</span></span></code></pre></div>
<p>That’s it! For analyzing multiome datasets, check out the arbalist
package.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Jean-Philippe Fortin, Aaron Lun, Natalie Fox, Jayaram Kancherla.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
